src/zone.js

```diff
@@ -429,6 +430,116 @@ Zone.prototype.createDisk = function(name, config, callback) {
   );
 };
 
+/**
+ * Create an instance group manager in this zone.
+ *
+ * @see [InstanceGroupManager Resource]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroupManagers#resource}
+ * @see [InstanceGroupManagers: insert API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroupManagers/insert}
+ *
+ * @param {string} name - Name of the instance group manger.
+ * @param {InstanceTemplate} instanceTemplate - Instance template to use for this instance group manager.
+ * @param {number} targetSize - Target number of running instances
+ * @param {object} options - See an
+ *     [InstanceGroupManager resource](https://cloud.google.com/compute/docs/reference/v1/instanceGroupManagers#resource).
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {InstanceGroup} callback.instanceGroupManager - The created
+ *     InstanceGroupManager object.
+ * @param {Operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * const Compute = require('@google-cloud/compute');
+ * const compute = new Compute();
+ * const zone = compute.zone('us-central1-a');
+ * const instanceTemplate = compute.instanceTemplate('my-instance-template');
+ * const targetSize = 10;
+ *
+ * function onCreated(
+ *   err,
+ *   instanceGroupManager,
+ *   targetSize,
+ *   operation,
+ *   apiResponse
+ * ) {
+ *   // `instanceGroupManager` is an InstanceGroupManager object.
+ *
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * }
+ *
+ * zone.createInstanceGroupManager(
+ *   'instance-group-manager-name',
+ *   instanceTemplate,
+ *   targetSize,
+ *   onCreated
+ * );
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * zone.createInstanceGroup('instance-group-manager-name', instanceTemplate).then(function(data) {
+ *   const instanceGroupManager = data[0];
+ *   const operation = data[1];
+ *   const apiResponse = data[2];
+ * });
+ */
+Zone.prototype.createInstanceGroupManager = function(
+  name,
+  instanceTemplate,
+  targetSize,
+  options,
+  callback
+) {
+  var self = this;
+
+  if (!common.util.isCustomType(instanceTemplate, 'InstanceTemplate')) {
+    throw new Error('An InstanceTemplate object is required.');
+  }
+
+  if (typeof targetSize !== 'number') {
```

Comment:
We can let the API throw instead of duplicating their requirements. They've been known to change on us, leaving us with stale code, so I prefer to let the API handle communicating the errors.